### ISSUE TITLE
Most fields "undefined" after upgrade to nextCloud 10

### ISSUE THREAD

**Issue by cdauth** [fcturner/passwords#294]
Sorry for writing again, I have another problem after upgrading to nextCloud 10.

Since the upgrade, almost all the fields in the password manager show “undefined”. The only field that shows its value is the website name.

Here is an example entry in the DOM:

```html
<tr attr_id="3" attr_website="Amazon" attr_address="undefined" attr_loginname="undefined" attr_pass="" attr_category="0" attr_notes="undefined" attr_datechanged="undefined" attr_strength="???" attr_length="undefined" attr_lower="undefined" attr_upper="undefined" attr_number="undefined" attr_special="undefined">
```

The corresponding entry from the `/apps/passwords/passwords` JSON object is this:

```json
{"id":3,"user_id":"cdauth","loginname":"","website":"Amazon","address":"","pass":false,"properties":false,"notes":false,"creation_date":"1970-01-01","deleted":false}
```

I definitely had login name, address and password set.

Looking at the database, the `loginname`, `address`, and `notes` columns are all completely empty! The `pass` column seems to contains some stuff, and `creation_date` is always `1970-01-01`.

**Comment by fcturner** [comment]
I never mind you writing :smile: I'm glad you're willing to share some problems so I can solve them for you and others.

> Looking at the database, the loginname, address, and notes columns are all completely empty! The pass column seems to contains some stuff, and creation_date is always 1970-01-01.

This is how it should be. Those columns aren't used anymore, and I will remove them in the next version. Now all those properties are saved encrypted in the `properties` column.

I don't know why your row fields are still empty. Nothing about a request timeout?

**Comment by cdauth** [comment]
Ah, that makes sense.

In the development console I see this exception:

```
Uncaught TypeError: text.replace is not a function
    at escapeHTML (passwords.js?v=dbd5a7d…:2374)
    at Notif.renderElement (notification.js?v=dbd5a7d…:112)
    at Object.addToUI (app.js?v=dbd5a7d…:288)
    at Object.<anonymous> (app.js?v=dbd5a7d…:152)
    at Function.each (jquery.min.js?v=dbd5a7d…:2)
    at app.js?v=dbd5a7d…:149
    at Object.<anonymous> (app.js?v=dbd5a7d…:344)
    at j (jquery.min.js?v=dbd5a7d…:2)
    at Object.fireWith [as resolveWith] (jquery.min.js?v=dbd5a7d…:2)
    at x (jquery.min.js?v=dbd5a7d…:4)
```

`text` is `6` (so a number) there, which is why `text.replace` does not exist. I don’t know if this is related to the issue. I will investigate further.

**Comment by cdauth** [comment]
I fixed that exception locally by converting `text` to a string, but that doesn't solve the issue. Everything is still undefined.

There are no request timeouts or failed requests.

Any clue how I could debug this further? Is there a way that I can decode that value of the properties column by hand so that I can investigate what is stored inside?

**Comment by cdauth** [comment]
I managed to solve the issue.

I was not aware that the passwords app is using the `passwordsalt` property from NextCloud’s `config.php` to encrypt its passwords. This salt got lost during my migration.

I restored the old passwordsalt and all my passwords are back.

I’m closing this now, but for the future it would be nice to see some message at least somewhere that indicates that the decryption of the passwords failed.
